src/Common/src/System/Collections/Generic/BidirectionalDictionary.cs

```diff
@@ -0,0 +1,47 @@
+// Licensed to the .NET Foundation under item1 or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Collections.Generic
+{
+    internal sealed class BidirectionalDictionary<T1, T2> : IEnumerable<Tuple<T1, T2>>
+    {
+        private readonly Dictionary<T1, T2> _forward;
+        private readonly Dictionary<T2, T1> _backward;
+
+        public BidirectionalDictionary(int capacity)
+        {
+            _forward = new Dictionary<T1, T2>(capacity);
+            _backward = new Dictionary<T2, T1>(capacity);
+        }
+
+        public void Add(T1 item1, T2 item2)
+        {
+            _forward.Add(item1, item2);
+            _backward.Add(item2, item1);
+        }
+
+        public bool TryGetForward(T1 item1, out T2 item2)
+        {
+            return _forward.TryGetValue(item1, out item2);
+        }
+
+        public bool TryGetBackward(T2 item2, out T1 item1)
+        {
+            return _backward.TryGetValue(item2, out item1);
+        }
+
+        public IEnumerator<Tuple<T1, T2>> GetEnumerator()
+        {
+            foreach (var kvp in _forward)
+            {
+                yield return new Tuple<T1, T2>(kvp.Key, kvp.Value);
+            }
+        }
```

Comment:
That's my point, though: it's not public, we don't need to worry about such things, and if this were to be made public the entire surface area (and likely implementation) would need to be revisited.  If we need an enumerator, we should keep it simple and do as I suggested just returning that of one of the underlying dictionaries.  If we don't need it, then we can remove it.